### ISSUE TITLE
Reconsider not handling line plots with markers

### ISSUE THREAD

**Issue by PeterPablo** [matlab2tikz/matlab2tikz#774]
Recently I converted a figure that contained a large number of data points that were plotted with markers. It took me a while to figure out that we [don't simplify line plots with markers](https://github.com/matlab2tikz/matlab2tikz/blob/develop/src/cleanfigure.m#L391).

My goal was to explicitly reduce the number of data points, even if this would change the visual appearance. This is currently not easily possible -- one would have to modify the marker property to `none`, apply cleanfigure and then again set the desired marker property.

What are your oppinions @matlab2tikz/developers?

**Comment by egeerardyn** [comment]
I think both approaches have their merits (I hope you hear me coming: we should have an option for it). Also, I think that `return` in that fragment should go: that's not a clear way to show the intent.

To illustrate my opinion, I think there are 3 kinds of plots:
 1. line plots (e.g. `plot(x,y, '-')`
 2. scatter plots (e.g `plot(x, y, 'o')`)
 3. line plots with markers (e.g. `plot(x, y, '-o')`)

To give a practical example:
```matlab
x = [1:5 8:10];
y = 0.5*x;

figure; hold all;
plot(x,y+1, '-' , 'DisplayName', '1: line only');
plot(x,y+2, 'o' , 'DisplayName', '2: marker only');
plot(x,y+3, 'o-', 'DisplayName', '3: marker & line');
xlabel('x');
ylabel('y');
legend('show', 'Location','northwest')
set(findobj(gca,'type','line'), 'LineWidth', 2)
```
<img width="469" alt="screen shot 2015-09-17 at 09 47 41" src="https://cloud.githubusercontent.com/assets/177360/9927720/35223cfc-5d21-11e5-8c75-16c181811f29.png">

For (1) we have everything under control already. In this case, the ideal simplification would be just the end points.

For (2), I think the same algorithms make little sense in general, since no lines are drawn, simplification should happen based on how densely a part of the view is covered by points, not whether removing a certain point will displace the connecting lines (since there are none). In my example, there is no such region, but e.g. the figure below certainly has such regions:

![check](https://cloud.githubusercontent.com/assets/177360/9814307/235db8e0-588d-11e5-9684-94684f22c850.png)

Let me explain my typical use for (3): I use those for plotting functions that I know to be smooth (=line part) or even piecewise linear, but where I have only observed some points (=markers). Here, I think it makes the most sense to also use a density-based criterion for removing points, not a line-based criterion (since removing a point/marker may convey different information).

**Comment by okomarov** [comment]
Clean figure pixelates only of there are markers. This reduces the number of points but in a way that visually it doesn't change.

Sent from my iPhone

On 17 Sep 2015, at 08:04, "Peter Pablo" <notifications@github.com<mailto:notifications@github.com>> wrote:


Recently I converted a figure that contained a large number of data points that were plotted with markers. It took me a while to figure out that we don't simplify line plots with markers<https://github.com/matlab2tikz/matlab2tikz/blob/develop/src/cleanfigure.m#L391>.

My goal was to explicitly reduce the number of data points, even if this would change the visual appearance. This is currently not easily possible -- one would have to modify the marker property to none, apply cleanfigure and then again set the desired marker property.

What are your oppinions @matlab2tikz/developers<https://github.com/orgs/matlab2tikz/teams/developers>?

**Comment by okomarov** [comment]
I think `cleanfigure`'s purpose is to reduce as much noise as possible while mantaining the original look of the plot. This mode fits the logic of a **conversion** utility as `m2t`.

If we want to change appearance of the plot, we should make the intent clear. This might clash with cleanfigure because the routine already does too many things and because we would need more optional flags to convey a clear meaning of its actions.

Let's take case (1), markers with line as an example of why the itent should be clear. Actually, the optimal simplification depends on the informative content of the markers. What if those markers are events happening along a time line? I would not want to erase them accidentally (and cleanfigure does not simplify them away, unless there are 1000 markers on 1 pixels).

What we would need is a `reducepoints()` routine with verbose flags, e.g. 'preserve', 'aggressive', and the numeric options for the resolution. This way, we can fine-tune the resolution of some tasks, while also toggling the mode. 

We could go embed the operative mode into `cleanfigure` and force e.g. `simplifyLine` on the first case (simplify away the markers too). However, what does 'aggressive' mean for other tasks in cleanfigure. Also, this probably won't solve the necessary sub-sampling for the scatter presented by Eagon, which needs a separate algo with its own (?) resolution parameters.

Basically, it's faster to tell the user: 
*"you know what you want to do; cleanfigure preserves appearance; if you want a more aggressive approach, first create the plot you want, and we'll make sure it comes out in tikz the same way, and efficiently"*

**Comment by miscco** [comment]
One should reference #824 and #825
